src/parsers/ldjson_feed.c

```diff
@@ -0,0 +1,364 @@
+
+static void
+ldjson_feed_parse_json_website (JsonNode *node, feedParserCtxtPtr ctxt)
+{
+	JsonNode	*logo;
+	const gchar	*tmp;
+
+	/* Try to extract subscription details (title, logo...)
+
+	   Example:
+
+	   {
+		"@context" : "http://schema.org",
+		"type" : "WebPage",
+		"url" : "https://www.meinestadt.de/",
+		"@type" : "Organization",
+		"name": "meinestadt.de GmbH",
+		"logo" : {
+			"@type" : "ImageObject",
+			"url" : "https://unternehmen.meinestadt.de/logos/meinestadt_de_logo.jpg",
+			"width" : 500,
+			"height" : 500
+		},
+		"sameAs" : [
+			"https://www.facebook.com/meinestadt.de",
+			"https://www.facebook.com/meinestadt.destellenmarkt",
+			"https://www.facebook.com/meinestadt.deapp",
+			"https://www.xing.com/company/meinestadt-de",
+			"https://www.linkedin.com/company/allesklar.com-ag",
+			"https://de.wikipedia.org/wiki/Meinestadt.de",
+			"https://www.youtube.com/user/meinestadtde",
+			"https://www.instagram.com/meinestadt.de/",
+			"https://twitter.com/meinestadt_de"
+		]
+	    }
+	*/
+
+	if (tmp = json_get_string (node, "name"))
+		node_set_title (ctxt->subscription->node, tmp);
+
+	if (tmp = json_get_string (node, "url"))
+		subscription_set_homepage (ctxt->subscription, tmp);
+
+	if (logo = json_get_node (node, "logo"))
+		if (tmp = json_get_string (logo, "url"))
+			metadata_list_set (&ctxt->subscription->metadata, "imageUrl", tmp);
+}
+
+static void
+ldjson_feed_parse_json_event (JsonNode *node, feedParserCtxtPtr ctxt)
+{
+	const gchar *tmp;
+	const gchar *eventStatus, *type;
+
+	ctxt->item = item_new ();
+
+	/* what we expect here:
+
+	1.) MusicEvent
+
+	{
+	"name": "Band name",
+	"image": "...",
+	"startDate": "2021-06-12T17:00:00+01:00",
+	"endDate": "2021-06-12",
+	"previousStartDate": "2020-06-17T16:00:00Z",
+	"url": "https://example.com",
+	"description": "detailed description",
+	"offers": {
+	   "url": "https://example.com/abc",
+	   "availabilityStarts": "2019-07-05T10:00:00Z",
+	   "priceCurrency": "GBP",
+	   "price": "89",
+	   "validFrom": "2019-07-05T10:00:00Z",
+	   "@type": "Offer"
+	},
+	"performer": [
+	  {
+	    "name": "Performer 1",
+	    "sameAs": "https://example.com/performer1",
+	    "@type": "MusicGroup"
+	  },
+	  {
+	    "name": "Performer 2",
+	    "sameAs": "https://example.com/performer2",
+	    "@type": "MusicGroup"
+	   }
+	],
+	"@type": "MusicEvent"
+	}
+	*/
+
+	/* eventStatus is a schema.org enum e.g. "https://schema.org/EventCancelled"
+	   let's strip the prefix and use the enum name as is */
+	type = json_get_string (node, "@type");
+	eventStatus = json_get_string (node, "eventStatus");
+	if (eventStatus && eventStatus == g_strstr_len (eventStatus, -1, "https://schema.org/Event"))
+		eventStatus += 24;
+	else
+		eventStatus = NULL;
+
+	if (tmp = json_get_string (node, "startDate")) {
+		// schema.org says startDate should be ISO8601, but RFC822
+		// is seen to often. So fuzzy match date format.
+		if (strstr (tmp, " "))
+			ctxt->item->time = date_parse_RFC822 (g_strdup (tmp));
```

Comment:
Sorry for being late, I haven't read everything, but there is a leak here. The `g_strdup` are not necessary.

Also the examples display `There were errors while parsing this feed!` for me. (But the items appears to work anyway)

Reply:
Thank you for taking the time to review!!! I'll fix the g_strdup().

About the error message: I'm working on a refactoring of the error handling, so this will be addressed soon.